File: scripts/style/ov_publication_style.R
OV_GROUP_COLORS <- c(
  Normal = "#80B1D3", Benign = "#8DD3C7", Borderline = "#FDB462", Malignant = "#FB8072",
  N = "#80B1D3", B = "#8DD3C7", BD = "#FDB462", M = "#FB8072"
)

OV_COHORT_COLORS <- c(
  "Discovery cohort" = "#8DD3C7",
  "Temporal same-centre validation cohort" = "#BEBADA"
)

OV_MODEL_COLORS <- c(
  "Baseline" = "#B3B3B3", "CA125-only" = "#B3B3B3",
  "Augmented" = "#FB8072", "Targeted model" = "#FB8072"
)

OV_DIRECTION_COLORS <- c(
  up = "#FB8072", down = "#80B1D3", positive = "#FB8072", negative = "#80B1D3",
  background = "#D9D9D9", missing = "#B3B3B3"
)

OV_PATHWAY_COLORS <- c(
  "Arginine/proline" = "#8DD3C7", "Phenylalanine" = "#FFFFB3",
  "Tryptophan" = "#BEBADA", "Steroid hormone" = "#FB8072"
)

OV_HEATMAP_BLUE <- "#2166AC"
OV_HEATMAP_WHITE <- "#F7F7F7"
OV_HEATMAP_RED <- "#B2182B"
OV_SIGNAL_RED <- "#FB8072"
OV_SIGNAL_BLUE <- "#80B1D3"
OV_BACKGROUND_GREY <- "#D9D9D9"
OV_TEXT_DARK <- "#000000"
OV_AXIS_GREY <- "#000000"

ov_theme <- function(base_size = 7, base_family = "Arial") {
  ggplot2::theme_classic(base_size = base_size, base_family = base_family) +
    ggplot2::theme(
      axis.line = ggplot2::element_line(linewidth = 0.35, colour = OV_AXIS_GREY),
      axis.ticks = ggplot2::element_line(linewidth = 0.35, colour = OV_AXIS_GREY),
      axis.text = ggplot2::element_text(colour = OV_TEXT_DARK, size = base_size - 0.5),
      axis.title = ggplot2::element_text(colour = OV_TEXT_DARK, size = base_size),
      legend.title = ggplot2::element_text(size = base_size - 0.5),
      legend.text = ggplot2::element_text(size = base_size - 0.7),
      plot.title = ggplot2::element_text(size = base_size + 0.5, face = "bold"),
      strip.text = ggplot2::element_text(size = base_size, face = "bold"),
      legend.background = ggplot2::element_blank(),
      panel.grid = ggplot2::element_blank()
    )
}

ov_save_pdf_png <- function(plot, file_stem, width = 5, height = 3.5, dpi = 600) {
  grDevices::pdf(paste0(file_stem, ".pdf"), width = width, height = height, family = "Arial", useDingbats = FALSE)
  print(plot)
  grDevices::dev.off()
  ggplot2::ggsave(paste0(file_stem, ".png"), plot, width = width, height = height, units = "in", dpi = dpi)
  if (requireNamespace("svglite", quietly = TRUE)) {
    svglite::svglite(paste0(file_stem, ".svg"), width = width, height = height)
    print(plot)
    grDevices::dev.off()
  }
}

File: 00_project_style/ov_publication_style.py
from __future__ import annotations

from pathlib import Path


GROUP_ORDER_FULL = ["Normal", "Benign", "Borderline", "Malignant"]
GROUP_ORDER_SHORT = ["N", "B", "BD", "M"]

GROUP_COLORS = {
    "Normal": "#80B1D3", "Benign": "#8DD3C7", "Borderline": "#FDB462", "Malignant": "#FB8072",
    "N": "#80B1D3", "B": "#8DD3C7", "BD": "#FDB462", "M": "#FB8072",
    "Normal controls": "#80B1D3", "Benign lesions": "#8DD3C7",
    "Borderline tumours": "#FDB462", "Malignant tumours": "#FB8072",
}

COHORT_COLORS = {
    "Discovery cohort": "#8DD3C7",
    "Temporal same-centre validation cohort": "#BEBADA",
    "Training": "#8DD3C7",
    "Validation": "#BEBADA",
}

MODEL_COLORS = {
    "Baseline": "#B3B3B3", "CA125-only": "#B3B3B3", "Augmented": "#FB8072",
    "Targeted model": "#FB8072", "Discovery cohort": "#8DD3C7",
    "Temporal same-centre validation cohort": "#BEBADA",
}

DIRECTION_COLORS = {
    "up": "#FB8072", "down": "#80B1D3", "positive": "#FB8072", "negative": "#80B1D3",
    "high": "#FB8072", "low": "#80B1D3", "background": "#D9D9D9",
    "missing": "#B3B3B3", "other": "#B3B3B3",
}

PATHWAY_COLORS = {
    "Arginine/proline": "#8DD3C7", "Phenylalanine": "#FFFFB3",
    "Tryptophan": "#BEBADA", "Steroid hormone": "#FB8072",
}

HEATMAP_BLUE = "#2166AC"
HEATMAP_WHITE = "#F7F7F7"
HEATMAP_RED = "#B2182B"
SIGNAL_RED = "#FB8072"
SIGNAL_BLUE = "#80B1D3"
BACKGROUND_GREY = "#D9D9D9"
TEXT_DARK = "#000000"
AXIS_GREY = "#000000"


def add_style_path() -> None:
    return None


def setup_matplotlib_style(mpl, sns=None, base_size: float = 7.0) -> None:
    mpl.rcParams.update({
        "font.family": "sans-serif", "font.sans-serif": ["Arial", "Helvetica", "DejaVu Sans", "sans-serif"],
        "font.size": base_size, "axes.labelsize": base_size, "axes.titlesize": base_size + 0.5,
        "xtick.labelsize": base_size - 0.5, "ytick.labelsize": base_size - 0.5,
        "legend.fontsize": base_size - 0.7, "legend.title_fontsize": base_size - 0.5,
        "axes.linewidth": 0.55, "xtick.major.width": 0.55, "ytick.major.width": 0.55,
        "xtick.major.size": 2.2, "ytick.major.size": 2.2, "pdf.fonttype": 42,
        "ps.fonttype": 42, "svg.fonttype": "none", "figure.dpi": 150, "savefig.dpi": 600,
        "text.color": TEXT_DARK, "axes.labelcolor": TEXT_DARK, "axes.edgecolor": AXIS_GREY,
        "xtick.color": TEXT_DARK, "ytick.color": TEXT_DARK,
    })
    if sns is not None:
        sns.set_style("ticks")


def save_publication_figure(fig, out_stem: Path, width: float, height: float, dpi: int = 600) -> None:
    fig.set_size_inches(width, height)
    out_stem = Path(out_stem)
    out_stem.parent.mkdir(parents=True, exist_ok=True)
    for ext in [".pdf", ".svg", ".png"]:
        fig.savefig(out_stem.with_suffix(ext), bbox_inches="tight", dpi=dpi)
